Mathlib/CategoryTheory/Limits/Shapes/Kernels.lean

```diff
@@ -1199,4 +1199,26 @@ instance (priority := 100) hasKernels_of_hasEqualizers [HasEqualizers C] : HasKe
 instance (priority := 100) hasCokernels_of_hasCoequalizers [HasCoequalizers C] :
     HasCokernels C where
 
+/-- The kernel of an arrow is natural. -/
+@[simps]
+noncomputable def kernelFunctor [HasKernels C] : Arrow C ⥤ C where
+  obj f := kernel f.hom
+  map {f g} u := kernel.lift _ (kernel.ι _ ≫ u.left) (by simp)
+
+/-- The cokernel of an arrow is natural. -/
+@[simps]
+noncomputable def cokernelFunctor [HasCokernels C] : Arrow C ⥤ C where
+  obj f := cokernel f.hom
+  map {f g} u := cokernel.desc _ (u.right ≫ cokernel.π _) (by simp [← Arrow.w_assoc u])
+
+/-- The kernel inclusion is natural. -/
+@[simps]
+def kernel.ιNatTrans [HasKernels C] : kernelFunctor (C := C) ⟶ Arrow.leftFunc where
```

Comment:
Then, it would be nice to show that the composition of `ιNatTrans` with `Arrow.leftToRight` is zero, and in another file (`Limits.FunctorCategory.Shapes.Kernels`?) show that this makes a limit kernel fork (using `evaluationJointlyReflectLimits` and the definitions in `Limits.Preserves.Shapes.Kernels`).

Reply:
I could define the kernel fork, but not show that it's a limit. Do you have a further hint?